modules/data.atmosphere/R/solar_angle.R

```diff
@@ -34,7 +34,8 @@ cos_solar_zenith_angle <- function(doy, lat, lon, dt, hr) {
 #' @return `numeric(1)` length of the solar day, in hours.
 #' @export
 equation_of_time <- function(doy) {
-  stopifnot(doy <= 366)
+  stopifnot(doy <= 367) #changed from 366 to 367 to account for leap years
```

Comment:
doest 366 already do that?

Reply:
So I found that the function used to calculate doy actually returns 366 for non-leap years and 367 for leap years